lib/core/plugins.js

```diff
@@ -98,18 +99,48 @@ Plugins.prototype.getPluginsFor = function(pluginType) {
   });
 };
 
-Plugins.prototype.getPluginsProperty = function(pluginType, property) {
+Plugins.prototype.getPluginsProperty = function(pluginType, property, sub_property) {
   let matchingPlugins = this.plugins.filter(function(plugin) {
     return plugin.has(pluginType);
   });
 
   let matchingProperties = matchingPlugins.map((plugin) => {
+    if (sub_property) {
+      return plugin[property][sub_property];
+    }
     return plugin[property];
   });
 
+  matchingProperties = matchingProperties.filter((property) => property);
+
   //return flattened list
   if (matchingProperties.length === 0) return [];
-  return matchingProperties.reduce((a,b) => { return a.concat(b); });
+  return matchingProperties.reduce((a,b) => { return a.concat(b); }) || [];
+};
+
+Plugins.prototype.runActionsForEvent = function(eventName, args, cb) {
+  if (typeof (args) === 'function') {
+    cb = args;
```

Comment:
Could you put `args = []` and that way you could remove the condition below? Or `args = null`? Looks like the function called by `plugin.call` already knows how to handle no args.